tests/drivers/timer/nrf_grtc_timer/src/main.c

```diff
@@ -37,4 +49,69 @@ ZTEST(nrf_grtc_timer, test_get_ticks)
 		     "Unexpected result %" PRId64 " (expected: %" PRId64 ")", ticks, exp_ticks);
 }
 
+ZTEST(nrf_grtc_timer, test_timer_count_in_compare_mode)
+{
+	int err;
+	int key;
+	uint64_t test_ticks = 0;
+	char user_data[] = "test_timer_abort_in_compare_mode test\n";
+	int32_t channel = z_nrf_grtc_timer_chan_alloc();
+
+	TC_PRINT("Allocated GRTC channel %d\n", channel);
+	if (channel < 0) {
+		TC_PRINT("Failed to allocate GRTC channel, chan=%d\n", channel);
+		ztest_test_fail();
+	}
+
+	compare_isr_call_counter = 0;
+	key = z_nrf_grtc_timer_compare_int_lock(channel);
+	test_ticks = z_nrf_grtc_timer_read() + K_MSEC(TIMER_COUNT_TIME_MS).ticks;
```

Comment:
You should use `z_nrf_grtc_timer_get_ticks()` instead as applications [are not supposed to inspect the internal data](https://github.com/zephyrproject-rtos/zephyr/blob/90d50cf14169f242ad14ff2262e3303921fea3be/include/zephyr/sys_clock.h#L56-L63) in `k_timeout_t`.

Reply:
Please leave an example i can't use this sucesfully.